src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,641 @@
+        @param tau    Time until you can write off the debt [sec]
+        @param culled Debt write off triggered (1 or 0)
+        @param tic    Timestamp when the pool is caged
+    */
+    struct Ilk {
+        ID3MPool pool;   // Access external pool and holds balances
+        ID3MPlan plan;   // How we calculate target debt
+        uint256  tau;    // Time until you can write off the debt [sec]
+        uint256  culled; // Debt write off triggered
+        uint256  tic;    // Timestamp when the d3m can be culled (tau + timestamp when caged)
+    }
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+    event File(bytes32 indexed ilk, bytes32 indexed what, address data);
+    event File(bytes32 indexed ilk, bytes32 indexed what, uint256 data);
+    event Wind(bytes32 indexed ilk, uint256 amount);
+    event Unwind(bytes32 indexed ilk, uint256 amount, uint256 fees);
+    event Reap(bytes32 indexed ilk, uint256 amt);
+    event Cage(bytes32 indexed ilk);
+    event Cull(bytes32 indexed ilk, uint256 ink, uint256 art);
+    event Uncull(bytes32 indexed ilk, uint256 wad);
+    event Quit(bytes32 indexed ilk, address indexed usr);
+    event Exit(bytes32 indexed ilk, address indexed usr, uint256 amt);
+
+    /**
+        @dev sets msg.sender as authed.
+        @param vat_     address of the DSS vat contract
+        @param daiJoin_ address of the DSS Dai Join contract
+    */
+    constructor(address vat_, address daiJoin_) {
+        vat = VatLike(vat_);
+        daiJoin = DaiJoinLike(daiJoin_);
+        TokenLike(DaiJoinLike(daiJoin_).dai()).approve(daiJoin_, type(uint256).max);
+        VatLike(vat_).hope(daiJoin_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /// @notice Modifier will revoke if msg.sender is not authorized.
+    modifier auth {
+        require(wards[msg.sender] == 1, "DssDirectDepositHub/not-authorized");
+        _;
+    }
+
+    /// @notice Mutex to prevent reentrancy on external functions
+    modifier lock {
+        require(locked == 0, "DssDirectDepositHub/system-locked");
+        locked = 1;
+        _;
+        locked = 0;
+    }
+
+    // --- Math ---
+    uint256 internal constant RAY = 10 ** 27;
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+    function _max(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x >= y ? x : y;
+    }
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x + (y - 1)) / y;
+    }
+
+    // --- Administration ---
+    /**
+        @notice Makes an address authorized to perform auth'ed functions.
+        @dev msg.sender must be authorized.
+        @param usr address to be authorized
+    */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
```

Comment:
```suggestion
    function rely(address usr) external auth {
        require(vat.live() == 1, "DssDirectDepositHub/no-rely-during-shutdown");
        wards[usr] = 1;
        emit Rely(usr);
    }
```

Reply:
The `hub` is a contract that can protect itself from malicious governance as it doesn't have any `auth` function that needs to be accessed after global settlement.
We should protect the `rely`/`deny` functions under the `vat.live` check to avoid relying on the `esm` keeper.
For `Pool`s we need a deeper discussion on what we want to do.
`Plan`s seem to be able to be self protected easily as well.

Reply:
Actually the `plan` is not necessary. After shutdown the `Hub` doesn't even use anything of the `plan`, so no need to protect this piece from malicious governance.

Reply:
Closing this, as malicious governance doesn't gain anything from `rely/deny`ing anyone and won't break this module either doing so.